exporter/splunkhecexporter/integration_test.go

```diff
@@ -0,0 +1,398 @@
+	// Stop and remove the container
+	fmt.Println("Stopping container")
+	err := splunkContCfg.container.Terminate(splunkContCfg.conCtx)
+	if err != nil {
+		logger.Printf("Error while preparing POST request")
+		panic(err)
+	}
+}
+
+func TestMain(m *testing.M) {
+	setup()
+
+	// Run the tests
+	code := m.Run()
+
+	teardown()
+	// Exit with the test result code
+	os.Exit(code)
+}
+
+func createInsecureClient() *http.Client {
+	// Create a custom transport with insecure settings
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	// Create the client with the custom transport
+	client := &http.Client{
+		Transport: tr,
+	}
+
+	return client
+}
+
+func startSplunk() {
+	ctx := context.Background()
+	splunkContCfg.conCtx = ctx
+
+	// Create a new container
+	req := testcontainers.ContainerRequest{
+		Image:        "splunk/splunk:latest",
+		ExposedPorts: []string{"8000/tcp", "8088/tcp", "8089/tcp"},
+		Env: map[string]string{
+			"SPLUNK_START_ARGS": "--accept-license",
+			"SPLUNK_HEC_TOKEN":  integrationtestutils.ViperGetConfigVariable("HEC_TOKEN"),
+			"SPLUNK_PASSWORD":   integrationtestutils.ViperGetConfigVariable("PASSWORD"),
+		},
+		WaitingFor: wait.ForLog("Ansible playbook complete, will begin streaming splunkd_stderr.log").WithStartupTimeout(2 * time.Minute),
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	splunkContCfg.container = container
+
+	if err != nil {
+		logger.Printf("Error while creating container")
+		panic(err)
+	}
+
+	// Get the container host and port
+	uiPort, err := container.MappedPort(ctx, "8000")
+	if err != nil {
+		logger.Printf("Error while getting port")
+		panic(err)
+	}
+
+	hecPort, err := container.MappedPort(ctx, "8088")
+	if err != nil {
+		logger.Printf("Error while getting port")
+		panic(err)
+	}
+	managementPort, err := container.MappedPort(ctx, "8089")
+	if err != nil {
+		logger.Printf("Error while getting port")
+		panic(err)
+	}
+	host, err := container.Host(ctx)
+	if err != nil {
+		logger.Printf("Error while getting host")
+		panic(err)
+	}
+
+	// Use the container's host and port for your tests
+	logger.Println("Splunk running at:", host, uiPort, hecPort, managementPort)
+	integrationtestutils.ViperSetConfigVariable("HOST", host)
+	integrationtestutils.ViperSetConfigVariable("UI_PORT", strconv.Itoa(uiPort.Int()))
+	integrationtestutils.ViperSetConfigVariable("HEC_PORT", strconv.Itoa(hecPort.Int()))
+	integrationtestutils.ViperSetConfigVariable("MANAGEMENT_PORT", strconv.Itoa(managementPort.Int()))
+}
+
+func prepareLogs() plog.Logs {
+	logs := plog.NewLogs()
+	logs.ResourceLogs().EnsureCapacity(1)
+	rl := logs.ResourceLogs().AppendEmpty()
+	rl.ScopeLogs().EnsureCapacity(1)
+
+	sl := rl.ScopeLogs().AppendEmpty()
+	sl.Scope().SetName("test")
+	sl.LogRecords().EnsureCapacity(1)
+
+	ts := pcommon.Timestamp(int64(0) * time.Millisecond.Nanoseconds())
```

Comment:
Why not just use `0` ? This feels like it was a typo but I want to make sure.

Reply:
yes, I will put 0 here